### ISSUE TITLE
PERMISSION_DENIED: The caller does not have permission

### ISSUE THREAD

**Issue by denisdnc** [GoogleCloudPlatform/functions-framework-nodejs#159]
Hy there!
How are you?

I'm developing a [Google Cloud Function][1] and it is great!
But since last week I'm getting an issue testing locally with [functions-framework-nodejs][2]

```
Unhandled rejection
Error: 7 PERMISSION_DENIED: The caller does not have permission
    at Object.callErrorFromStatus (/home/user/project/node_modules/@grpc/grpc-js/build/src/call.js:31:26)
    at Object.onReceiveStatus (/home/user/project/node_modules/@grpc/grpc-js/build/src/client.js:176:52)
    at Object.onReceiveStatus (/home/user/project/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:342:141)
    at Object.onReceiveStatus (/home/user/project/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:305:181)
    at Http2CallStream.outputStatus (/home/user/project/node_modules/@grpc/grpc-js/build/src/call-stream.js:117:74)
    at Http2CallStream.maybeOutputStatus (/home/user/project/node_modules/@grpc/grpc-js/build/src/call-stream.js:156:22)
    at Http2CallStream.endCall (/home/user/project/node_modules/@grpc/grpc-js/build/src/call-stream.js:142:18)
    at Http2CallStream.handleTrailers (/home/user/project/node_modules/@grpc/grpc-js/build/src/call-stream.js:274:14)
    at ClientHttp2Stream.emit (events.js:198:13)
    at ClientHttp2Stream.EventEmitter.emit (domain.js:466:23)
Error: Process exited with code 16
    at process.on.code (/home/user/project/node_modules/@google-cloud/functions-framework/build/src/invoker.js:271:38)
    at process.emit (events.js:198:13)
    at process.EventEmitter.emit (domain.js:448:20)
    at process.exit (internal/process/per_thread.js:168:15)
    at Object.logAndSendError (/home/user/project/node_modules/@google-cloud/functions-framework/build/src/logger.js:37:9)
    at process.on.err (/home/user/project/node_modules/@google-cloud/functions-framework/build/src/invoker.js:268:22)
    at process.emit (events.js:198:13)
    at process.EventEmitter.emit (domain.js:448:20)
    at emitPromiseRejectionWarnings (internal/process/promises.js:140:18)
    at process._tickCallback (internal/process/next_tick.js:69:34)
error Command failed with exit code 16.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

It occurs at the end of the execution of the function.

Any clue of what it is?

Version: "@google-cloud/functions-framework": "^1.6.0"

Thanks!

  [1]: https://cloud.google.com/functions
  [2]: https://github.com/GoogleCloudPlatform/functions-framework-nodejs

**Comment by denisdnc** [comment]
I tried to isolate some parts of the code and it seems to be [LogginWinston](https://github.com/googleapis/nodejs-logging-winston) it's causing the error, when i comment the "loggingWinston" on the creation of the logger, it works just fine. I dont know if there's a relation with the function framework.

Thanks!

**Comment by grant** [comment]
Hi @denisdnc, I'm not sure this is a Functions Framework problem.

Can you provide exact instructions on how to reproduce this error? Are you sure that you can't reproduce this error without using the Functions Framework?

It might be an issue with `@google-cloud/logging-winston`.

**Comment by denisdnc** [comment]
Hi there! I think i found out the problem, it's not really a problem, but sure it seems to in the @google-cloud/logging-winston.

It was missing a permission for the Logging API on the service account.
I think just the message it wasn't much clear.

But the code is right here:

https://github.com/denisdnc/cloud-functions-example

Set a service account credentials path:

```sh
export GOOGLE_APPLICATION_CREDENTIALS="/home/user/credential.json"
```

And start the app:
```sh
npx @google-cloud/functions-framework --target=execute
```

Do the request on http://localhost:8080/

Then should log the problem above.

Thank you guys for the support!

**Comment by grant** [comment]
Good to hear! If others have this problem, I hope they can search the issues and find the solution.

**Comment by edisonspencer** [comment]
Sorry for the long delay @denisdnc .
I already had the `Logs Writer` and added the ` Cloud Functions Admin` as per your screenshot, but no luck. Do you know if you need to download a new set of keys when changing the permissions attached to a user? I would expect not, I would expect it to take effect almost immediately.

**Comment by denisdnc** [comment]
It should take effect immediately, it worked for my service account. Maybe try to download a new credential just in case.

Sorry about the delay!

Thanks!

**Comment by edisonspencer** [comment]
No worries, still no luck. I must be doing something wrong. Tempted in deleting the project and setting up a new one. There must be something I've missed across the two if them.
Thanks anyway 👍

**Comment by edisonspencer** [comment]
Just to let you know that I managed to get it to work... Just leaving it here for future reference of others that might be having the same issue.
I ended up installing the gcloud CLI and while running some commands, I believe it was `gcloud config set project 'project-id'`, it must have done something, to the point that I no longer had my logger IAM account when I was looking on the IAM section of the GCP website. Once I set it up with the `Logger Writer` permission and re-run the project, I was able to see the logs.
